test/fake_report_server.ts

```diff
@@ -0,0 +1,97 @@
+import { Server, Socket } from 'net'
+import express from 'express'
+import { pipeline, Writable } from 'stream'
+import http from 'http'
+import { promisify } from 'util'
+
+type Callback = (err?: Error | null) => void
+
+/**
+ * Fake implementation of the same report server that backs Cucumber Reports
+ * (https://messages.cucumber.io). Used for testing only.
+ */
+export default class FakeReportServer {
+  private readonly sockets = new Set<Socket>()
+  private readonly server: Server
+  private receivedBodies = Buffer.alloc(0)
+  public receivedHeaders: http.IncomingHttpHeaders = {}
+
+  constructor(private readonly port: number) {
+    const app = express()
+
+    app.put('/s3', (req, res) => {
+      this.receivedHeaders = { ...this.receivedHeaders, ...req.headers }
+
+      // eslint-disable-next-line @typescript-eslint/no-this-alias
+      const server = this
```

Comment:
Why is `server` needed? Couldn't `write` be defined with an arrow function?

Reply:
Fixed in 508c876b05bee840466de3845becf87e27f79001